### ISSUE TITLE
Capability for deactivating the display of local variables in stack traces.

### ISSUE THREAD

**Issue by ruaultadrien** [iterative/dvc#9232]
Hello,

I am very often facing the issue that the local variables displayed in my stacktraces when using DVC take a huuuge amount of place in my terminal. It makes things very complicated to navigate and I would like to be able to disable the display of these local variables. So there are two reasons actually:

- The first one as I mentioned is that it takes a lot of place in the terminal which hides the different steps of the stacktrace.
- The second one is that I am actually almost never using the local variables displayed in the stack trace because it is just not convenient. When you are working with dataframes, it will only show you part of them which is most often not enough to draw any conclusions.

Basically I simply some times would like to go back to a normal python stack track because this is just exactly what I need, not the locals which I could get in a more usable way, in a debugger for example. I prefer the former behavior of DVC's stack trace which were more readable in my opinion.

So basically I was wondering if we could work towards having a settings that allows us to disable local variables in stack traces 🙏 please please please.

A workaround that was used by some colleague is to simply uninstall `rich` 😅 

I still love you and your work, but I could love you even more!

Best,

Adrien

**Comment by daavoo** [comment]
Hi! thanks for the report.

> I am very often facing the issue that the local variables displayed in my stacktraces when using DVC take a huuuge amount of place in my terminal. It makes things very complicated to navigate and I would like to be able to disable the display of these local variables.

I am not sure I understand what behavior you are referring to.
Could you share some reproduction of the setup you follow in the env you encounter the behavior or an example output?

**Comment by skshetry** [comment]
DVC does not add any traceback handlers. It might be your own script that's running `rich.install()` somewhere.
